include/Spectra/JDSymEigsDPR.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2020 Netherlands eScience Center <f.zapata@esciencecenter.nl>
+//
+// This Source Code Form is subject to the terms of the Mozilla
+// Public License v. 2.0. If a copy of the MPL was not distributed
+// with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef SPECTRA_JD_SYM_EIGS_DPR_H
+#define SPECTRA_JD_SYM_EIGS_DPR_H
+
+#include <Eigen/Dense>
+#include "JDSymEigsBase.h"
+#include "Util/SelectionRule.h"
+
+namespace Spectra {
+
+///
+/// \ingroup EigenSolver
+///
+/// This class implement the DPR correction for the Davidson algorithms.
+/// The algorithms in the Davidson family only differ in how the correction
+/// vectors are computed and optionally in the initial orthogonal basis set.
+///
+/// the DPR correction compute the new correction vector using the following expression:
+/// \f[ correction = -(\boldsymbol{D} - \rho \boldsymbol{I})^{-1} \boldsymbol{r} \f]
+/// where
+/// \f$D\f$ is the diagonal of the target matrix, \f$\rho\f$ the Ritz eigenvalue,
+/// \f$I\f$ the identity matrix and \f$r\f$ the residue vector.
+///
+
+template <typename Scalar, typename OpType>
+class JDSymEigsDPR : public JDSymEigsBase<Scalar, OpType>
+{
+private:
+    using Index = Eigen::Index;
+    using Matrix = Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic>;
+    using Vector = Eigen::Matrix<Scalar, Eigen::Dynamic, 1>;
+
+    Vector diagonal_ = Vector::Zero(this->operator_dimension_);
```

Comment:
resize the vector, when you build the operator.

Reply:
not here.

Reply:
@JensWehner I don't understand what I need to resize here. The diagonal has the same size than the square underlying matrix.

Reply:
I would agree with Felipe here ...

Reply:
I would just leave it uninitilized until you fill it, was what I meant to say.

Reply:
done